lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -334,6 +370,113 @@ def format(tag, time, record)
       [tag, time, record].to_msgpack
     end
 
+    # Compute the monitored resource and common labels shared by a collection of
+    # entries.
+    def compute_group_resource_and_labels(tag)
+      # Note that we assume that labels added to group_common_labels below are
+      # not 'service' labels (i.e. we do not call extract_resource_labels
+      # again).
+      group_resource = @resource.dup
+      group_common_labels = @common_labels.dup
+
+      if @running_cloudfunctions
+        # If the current group of entries is coming from a Cloud Functions
+        # function, the function name can be extracted from the tag.
+        match_data = @cloudfunctions_tag_regexp.match(tag)
+        if match_data
+          # Resource type is set to Cloud Functions only for logs actually
+          # coming from a function, otherwise we leave it as Container.
+          group_resource.type = CLOUDFUNCTIONS_RESOURCE_TYPE
+          group_resource.labels['region'] = @gcf_region
+          group_resource.labels['function_name'] =
+            decode_cloudfunctions_function_name(
+              match_data['encoded_function_name'])
+          # Move GKE container labels from the MonitoredResource to the
+          # LogEntry.
+          instance_id = group_resource.labels.delete('instance_id')
+          group_common_labels["#{CONTAINER_SERVICE}/cluster_name"] =
+            group_resource.labels.delete('cluster_name')
+          group_common_labels["#{CONTAINER_SERVICE}/instance_id"] =
+            instance_id
+          group_common_labels["#{COMPUTE_SERVICE}/resource_id"] = instance_id
+          group_common_labels["#{COMPUTE_SERVICE}/zone"] =
+            group_resource.labels.delete('zone')
+        end
+      end
+      if group_resource.type == CONTAINER_RESOURCE_TYPE &&
+         @compiled_kubernetes_tag_regexp
+        # Container logs in Kubernetes are tagged based on where they came
+        # from, so we can extract useful metadata from the tag.
+        # Do this here to avoid having to repeat it for each record.
+        match_data = @compiled_kubernetes_tag_regexp.match(tag)
+        if match_data
+          group_resource.labels['container_name'] =
+            match_data['container_name']
+          %w(namespace_name pod_name).each do |field|
+            group_common_labels["#{CONTAINER_SERVICE}/#{field}"] =
+              match_data[field]
+          end
+        end
+      end
+
+      # Freeze the per-request state. Any further changes must be made on a
+      # per-entry basis.
+      group_resource.freeze
+      group_resource.labels.freeze
+      group_common_labels.freeze
+
+      [group_resource, group_common_labels]
+    end
+
+    # Extract entry resource and common labels that should be applied to
+    # individual entries from the group resource and group common labels.
```

Comment:
Nice catch! Yeah, it turned out the group common labels were not being used, so I removed them. Just removed it from the comment as well.